domino/http_request_manager.py

```diff
@@ -23,17 +27,31 @@ class _HttpRequestManager:
     def __init__(self, auth: AuthBase):
         self.auth = auth
         self._logger = logging.getLogger(__name__)
-        self.request_session = requests.Session()
-        self.request_session.headers["User-Agent"] = f"python-domino/{__version__}"
+        self.request_session = self._set_session()
 
         if isinstance(self.auth, _SessionInitializer):
-            self.auth.__initialize__(self.request_session)
+            self.auth.__initialize__(self._set_session())
+
+    def _set_session(self):
+        """
+        Initialize a request session with retry to help manage connections drop.
+        """
+        self.session = requests.Session()
+        retries = Retry(
+            total=int(os.getenv("DOMINO_MAX_RETRIES", str(R_SESSION_MAX_RETRIES))),
+            backoff_factor=1,  # retry seconds
+            status_forcelist=[502, 503, 504],
```

Comment:
A couple of quick checks: Does this also retry on the following?:
- 429 status codes, ideally awaiting the necessary time specified in the headers
- 408 status codes (not sure if we ever use them, but generally retrying on those makes sense)
- socket-level errors such as ECONNRESET and ECONNTIMEOUT

Reply:
408 does make sense, and we can include that, 
ECONNRESET and ECONNTIMEOUT and being handled, 
429, This can be handled by the end user.